src/providers.ts

```diff
@@ -152,6 +152,13 @@ export class SimpleLanguageInfoProvider {
   }
 }
 
+/**
+ * Specify a timeout when tokenizing a line to prevent a long line from locking
+ * up the main thread. Note this is used in VS Code:
+ * https://github.com/microsoft/vscode/blob/504c5a768a001b2099dd2b44e9dc39e10ccdfb56/src/vs/workbench/services/textMate/common/TMTokenization.ts#L39
+ */
+const TOKENIZE_TIMEOUT_MS = 500;
```

Comment:
Should this timeout be configurable?

Reply:
Good idea: will address in a follow-up diff!